pkg/utils/utils.go

```diff
@@ -70,3 +72,35 @@ func DecompressGZIP(compressedData []byte) (decompressedData []byte, err error)
 
 	return decompressedData, nil
 }
+
+// TODO: check if it's worth trying to compress the data prior to compressing it (e.g: using entropy algorithms)
+// This would save some CPU usage client side
+// TODO: have telemetry data type be passed in as a parameter to further check if we should compress data and which algorithm to use (e.g: deflate or gzip)
+func ShouldCompress(data []byte) (resultData []byte, compression *string, err error) {
+	// 'compression' is inserted as a sql.NullString, hence it is returned as a nullable string
+	var nilStr *string = nil
```

Comment:
You could probably just return nil rather than having to define a variable, or cast the nil as a string when returning it...